### PR TITLE
ZCS-5624 Bug 109017 - Non-persistent XSS - Web Client (HTML Search) [CWE-79]

### PR DESCRIPTION
**Pull request by silentsakky**
- prevent xss by sanitizing id before adding to dom

### CODE REVIEW

File: WebRoot/WEB-INF/tags/button.tag [hunk 1]
```diff
@@ -37,9 +37,9 @@
 </c:choose>
 <c:if test="${app:boolean(width)}"><c:set var="width" value="${width}"/></c:if>
 <c:if test="${not empty src}">
-    <td height="100%" nowrap="nowrap" valign="middle"><input name="${name}" type="image" src="<app:imgurl value='${src}' />" ${extra} <c:if test="${not empty id}">id="I${zm:cook(id)}"</c:if> <c:if test="${app:boolean(disabled)}">disabled </c:if> <c:if test="${not empty tooltip}">alt="${fn:escapeXml(tooltip)}" title="${fn:escapeXml(tooltip)}"</c:if> <c:if test="${not empty clazz}">class='${clazz}'</c:if>></td>
+    <td height="100%" nowrap="nowrap" valign="middle"><input name="${name}" type="image" src="<app:imgurl value='${zm:cook(src)}' />" ${zm:cook(extra)} <c:if test="${not empty id}">id="I${zm:cook(id)}"</c:if> <c:if test="${app:boolean(disabled)}">disabled </c:if> <c:if test="${not empty tooltip}">alt="${fn:escapeXml(tooltip)}" title="${fn:escapeXml(tooltip)}"</c:if> <c:if test="${not empty clazz}">class='${zm:cook(clazz)}'</c:if>></td>
```
Comment (plobbes) [inline WebRoot/WEB-INF/tags/button.tag]: I'd also update `<input name="${name}"...`

Comment (silentsakky) [reply WebRoot/WEB-INF/tags/button.tag]: Fixed
